woodwork/tests/accessor/test_table_accessor.py

```diff
@@ -817,30 +821,109 @@ def test_accessor_repr(sample_df, sample_column_names, sample_inferred_logical_t
+    assert isinstance(copied_df.ww.schema, Schema)
+    assert copied_df.ww.schema == schema_df.ww.schema
+
+    pd.testing.assert_frame_equal(to_pandas(schema_df), to_pandas(copied_df))
+
+    warning = 'DataFrame created by astype is not valid for the given typing information:\n dtype mismatch for column id between DataFrame dtype, string, and LogicalType dtype, Int64.\n Please initialize Woodwork with DataFrame.ww.init'
+    with pytest.warns(CannotInitSchemaWarning, match=warning):
+        new_df = schema_df.ww.astype({'id': 'string'})
+    assert new_df['id'].dtype == 'string'
+    assert new_df.ww.schema is None
+    assert schema_df.ww.schema is not None
+
+
+def test_dataframe_methods_on_accessor_inplace(sample_df):
+    xfail_dask_and_koalas(sample_df)
+
+    schema_df = sample_df.copy()
+    schema_df.ww.init(name='test_schema')
+
+    df_pre_sort = schema_df.copy()
+
+    schema_df.ww.sort_values(['full_name'], inplace=True)
+    assert schema_df.ww.name == 'test_schema'
+
+    pd.testing.assert_frame_equal(to_pandas(schema_df), to_pandas(df_pre_sort.sort_values(['full_name'])))
+
+    warning = "Operation performed by rename has invalidated the Woodwork typing information:\n The following columns in the DataFrame were missing from the typing information: {'new_name'}.\n Please reinitialize Woodwork with DataFrame.ww.init"
+    with pytest.warns(SchemaInvalidatedWarning, match=warning):
+        schema_df.ww.rename({'id': 'new_name'}, inplace=True, axis=1)
+    assert 'new_name' in schema_df.columns
+    assert schema_df.ww.schema is None
+
+
+def test_dataframe_methods_on_accessor_returning_series(sample_df):
+    xfail_dask_and_koalas(sample_df)
+
+    schema_df = sample_df.copy()
+    schema_df.ww.init(name='test_schema')
+
+    dtypes = schema_df.ww.dtypes
+
+    assert schema_df.ww.name == 'test_schema'
+    pd.testing.assert_series_equal(dtypes, schema_df.dtypes)
+
+    memory = schema_df.ww.memory_usage()
+    assert schema_df.ww.name == 'test_schema'
+    pd.testing.assert_series_equal(memory, schema_df.memory_usage())
+
+    warning = "Operation performed by pop has invalidated the Woodwork typing information:\n The following columns in the typing information were missing from the DataFrame: {'id'}.\n Please reinitialize Woodwork with DataFrame.ww.init"
+    with pytest.warns(SchemaInvalidatedWarning, match=warning):
+        schema_df.ww.pop('id')
+    assert 'id' not in schema_df.columns
+    assert schema_df.ww.schema is None
+
+
+def test_dataframe_methods_on_accessor_other_returns(sample_df):
+    xfail_dask_and_koalas(sample_df)
+    schema_df = sample_df.copy()
+    schema_df.ww.init(name='test_schema')
+
+    shape = schema_df.ww.shape
+
+    assert schema_df.ww.name == 'test_schema'
+    assert shape == schema_df.shape
 
-    different_dtype_df = schema_df.astype({'id': 'Int64'}, copy=True)
-    assert not _is_valid_schema(different_dtype_df, schema)
+    keys = schema_df.ww.keys()
```

Comment:
Move this immediately before the assertion where it is used, or update the assertion to use `schema_df.ww.keys()` directly to make this a bit more readable.

Reply:
changing to not set the `keys` variable at all and just use it directly